### PR TITLE
Input styling changes, Form component addition

### PR DESCRIPTION
**Pull request by kylesuss**
The `Input` element has new styling, per the designs here:

https://projects.invisionapp.com/d/main#/console/20597416/432876968/preview

In addition to that, I have added a "lightweight"`Form` component that does not make any assumptions about how you manage form state (values, onChange, etc.), though it does encode some UX patterns for how we handle error messages & validate form fields. The intent was to make this extremely flexible, but create a set of behaviors that we will want on form across our properties based on the design language. I added some example for how to manage state in the stories, but they are mostly meant for documentation/example purposes.

The API is backwards compatible, so I am setting this to a `minor` release.

PS this will be the first design system component I added w/ TypeScript so let me know what I can do to improve that 👍

### CODE REVIEW

File: src/components/tooltip/Tooltip.js [hunk 1]
```diff
@@ -80,14 +87,14 @@ export function Tooltip({
 }) {
   return (
     <TooltipWrapper hasChrome={hasChrome} data-placement={placement} ref={tooltipRef} {...props}>
-      {hasChrome && <Arrow data-placement={placement} ref={arrowRef} {...arrowProps} />}
+      <Arrow isVisible={hasChrome} data-placement={placement} ref={arrowRef} {...arrowProps} />
```
Comment (winkerVSbecks) [inline src/components/tooltip/Tooltip.js:90]: Thank you for adding this one!

File: src/components/Form.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useState } from 'react';
+
+interface Field {
+  id: string;
+  validationError: (value: string) => string | void;
+  Component: (props: any) => JSX.Element;
```
Comment (winkerVSbecks) [inline src/components/Form.tsx:6]: I believe `ComponentType` might be more appropriate here. @ndelangen wdyt?

```ts
interface Field<T> {
  id: string;
  validationError: (value: string) => string | void;
  Component: React.ComponentType<T>
}
```

**Review by winkerVSbecks — COMMENTED** [review]
Not sure about that one typing but, other LGTM

File: src/components/tooltip/Tooltip.js (hunk 1 above)
Comment (kylesuss) [reply src/components/tooltip/Tooltip.js:90]: No prob -- did you run into this issue as well?

Comment (winkerVSbecks) [reply src/components/tooltip/Tooltip.js:90]: yea, while adding the search dropdown on the blog

File: src/components/Form.tsx (hunk 2 above)
Comment (ndelangen) [reply src/components/Form.tsx:6]: yeah!

**Review by ndelangen — APPROVED** [review]

**Comment by kylesuss** [comment]
@ghengeveld I reworked this w/ the pattern we discussed on the call today. I think it ends up being more flexible, as you say. LMK what you think.

File: src/components/FormErrorState.tsx [hunk 3]
```diff
@@ -56,7 +57,7 @@ export const PureFormErrorState = ({
   return children({ getFormErrorFieldProps });
 };
 
-export const FormErrorState = (props: FormErrorStateProps) => {
+export const FormErrorState = ({ didAttemptSubmission = false, ...rest }: FormErrorStateProps) => {
```
Comment (ndelangen) [inline src/components/FormErrorState.tsx:60]:
```suggestion
export const FormErrorState: FunctionComponent<FormErrorStateProps> = ({ didAttemptSubmission = false, ...rest }) => {
```

Comment (kylesuss) [reply src/components/FormErrorState.tsx]: Nice one. Thanks.

**Review by ndelangen — APPROVED** [review]

**Review by ghengeveld — APPROVED** [review]

**Comment by ndelangen** [comment]
👏

**Comment by kylesuss** [comment]
:rocket: PR was released in [`v5.2.0`](https://github.com/storybookjs/design-system/releases/tag/v5.2.0) :rocket:

**Pull request merged**